src/doc/trpl/concurrency.md

```diff
@@ -0,0 +1,343 @@
+% Concurrency
+
+Concurrency and parallelism are incredibly important topics in computer
+science, and are also a hot topic in industry today. Computers are gaining more
+and more cores, yet many programmers aren't prepared to fully utilize them.
+
+Rust's memory safety features also apply to its concurrency story too. Even
+concurrent Rust programs must be memory safe. Rust's type system is up to the
```

Comment:
You might want to go ahead and mention "data races" here to be clear that it's a part of the "memory safety" guarantees Rust provides.